subsys/spm/Kconfig

```diff
@@ -48,6 +48,16 @@ menuconfig IS_SPM
 	select TRUSTED_EXECUTION_SECURE
 
 if IS_SPM
+
+config SPM_DEBUG
```

Comment:
Do we need this entry to be visible to the user at all; If we can turn this on automatically, why does the user need to see this?

We could omit the `prompt` altogether if the optimization level is not zero or debug, or if the `DEBUG` symbol is defined. Otherwise we could show the entry (which would be enabled), and let the user decide what to do with it. In that case I think we could get rid of the `#warning` directive.

Reply:
I disagree, it should be configurable. Mostly due to the UI limitations in menuconfig. 

Another point could be made with regards to the warning directive. People have dropped debug releases in to the wild before. It's better to warn when they are trying to do something catastrophical.

Reply:
Why @sigvartmh, is there any situation where the developer wants to have no optimizations/debug builds and keep the lock bit set (and thus be unable to debug)? Vice-versa, is there any situation where they are building a production binary and want to keep the lock bit un-set?